tests/integration/golang/aim/metric/search_test.go

```diff
@@ -34,97 +41,748 @@ func (s *SearchMetricsTestSuite) SetupTest() {
+			request: request.SearchMetricRequest{
+				Query: `((metric.name == "TestMetric1") or (metric.name == "TestMetric2") or (metric.name == "TestMetric3")) and metric.last == 1.6 and run.name != "TestRun1"`,
+			},
+
+			metrics: []*models.LatestMetric{
+				metric2Run3,
+			},
+		},
+		{
+			name: "SearchMetricLastRunNameOperationNotEqualsAndNotEqual",
+			request: request.SearchMetricRequest{
+				Query: `((metric.name == "TestMetric1") or (metric.name == "TestMetric2") or (metric.name == "TestMetric3")) and metric.last != 1.6 and run.name != "TestRun1"`,
+			},
+
+			metrics: []*models.LatestMetric{
+				metric1Run2,
+				metric2Run2,
+				metric3Run2,
+				metric1Run3,
+			},
+		},
+		{
+			name: "SearchMetricLastRunNameOperationGraterAndNotEqual",
+			request: request.SearchMetricRequest{
+				Query: `((metric.name == "TestMetric1") or (metric.name == "TestMetric2") or (metric.name == "TestMetric3")) and metric.last > 1.6 and run.name != "TestRun1"`,
+			},
+
+			metrics: []*models.LatestMetric{
+				metric2Run2,
+				metric3Run2,
+			},
+		},
+		{
+			name: "SearchMetricLastRunNameOperationLessAndNotEqual",
+			request: request.SearchMetricRequest{
+				Query: `((metric.name == "TestMetric1") or (metric.name == "TestMetric2") or (metric.name == "TestMetric3")) and metric.last < 1.6 and run.name != "TestRun1"`,
+			},
+
+			metrics: []*models.LatestMetric{
+				metric1Run2,
+				metric1Run3,
+			},
+		},
+		{
+			name: "SearchMetricLastRunNameOperationGraterOrEqualAndNotEqual",
+			request: request.SearchMetricRequest{
+				Query: `((metric.name == "TestMetric1") or (metric.name == "TestMetric2") or (metric.name == "TestMetric3")) and metric.last >= 1.6 and run.name != "TestRun1"`,
+			},
+
+			metrics: []*models.LatestMetric{
+				metric2Run2,
+				metric3Run2,
+				metric2Run3,
+			},
+		},
+		{
+			name: "SearchMetricLastRunNameOperationLessOrEqualAndNotEqual",
+			request: request.SearchMetricRequest{
+				Query: `((metric.name == "TestMetric1") or (metric.name == "TestMetric2") or (metric.name == "TestMetric3")) and metric.last <= 1.6 and run.name != "TestRun1"`,
+			},
+
+			metrics: []*models.LatestMetric{
+				metric1Run2,
+				metric1Run3,
+				metric2Run3,
+			},
+		},
+		{
+			name: "SearchMetricComplexQuery",
+			request: request.SearchMetricRequest{
+				Query: `((metric.name == "TestMetric1") or (metric.name == "TestMetric2")) and metric.last_step >= 1 and (run.name.endswith("2") or re.match("TestRun1", run.name)) and (metric.last < 1.6) and run.duration > 0`,
+			},
+
+			metrics: []*models.LatestMetric{
+				metric1Run2,
+			},
 		},
 	}
 	for _, tt := range tests {
 		s.T().Run(tt.name, func(T *testing.T) {
-			data, err := s.client.DoStreamRequest(
+			var resp []byte
+			query, err := urlquery.Marshal(tt.request)
+			assert.Nil(s.T(), err)
+			resp, err = s.client.DoStreamRequest(
 				http.MethodGet,
-				fmt.Sprintf("/runs/search/metric?%s", tt.query),
+				fmt.Sprintf(`/runs/search/metric?%s`, query),
```

Comment:
please use proper query string encoding by building a `net.url.URL` struct

Reply:
```
$ grep -rE 'fmt\.Sprintf\("/.*\?' tests                    
tests/integration/golang/aim/experiment/get_experiment_runs_test.go:            fmt.Sprintf("/experiments/%d/runs?limit=4&offset=%s", *experiment.ID, runs[8].ID),
tests/integration/golang/aim/run/search_test.go:                                fmt.Sprintf("/runs/search/run?%s", query),
tests/integration/golang/aim/run/archive_batch_test.go:                         fmt.Sprintf("/runs/archive-batch?archive=%s", tt.archiveParam),
tests/integration/golang/aim/metric/search_test.go:                             fmt.Sprintf("/runs/search/metric?%s", tt.query),
```

Reply:
yep, we can improve it a little bit and pass for example object, which could contain `body` parameter or `query` parameters and under the hood client will attach/serialise body or build query string.

Reply:
I'll create an issue to deal with it

Reply:
sure! I can create a concept then.

Reply:
https://github.com/G-Research/fasttrackml/issues/393 — assigned to you @dsuhinin

Reply:
if you start working on this, please target `features/namespaces`!

Reply:
yep yep. already!